### ISSUE TITLE
[templates] How to handle expressions in tables

### ISSUE THREAD

**Issue by justinfagnani** [WICG/webcomponents#703]
Currently, this won't parse in a way that's useful for dynamically creating tables:

```html
<template>
  <table>
    {{rows}}
  </table>
</template>
```

because the `{{rows}}` text will be fostered to before the table.

A couple of options are:

  * Offer a fallback syntax that can be parsed inside a table, like `<!--{{rows}}-->` or `<?{{rows}}>`
  * Change the parsing rules inside `<template>` so that disallowed content is not fostered.

**Comment by jakearchibald** [comment]
Parser changes can be limited to cases where the first non-whitespace chars are `{{`

**Comment by justinfagnani** [comment]
If there are parser changes, I think we might still need a syntax fallback for older browsers using a template parts polyfill.

**Comment by matthewp** [comment]
@justinfagnani Am I reading this right, is the idea that you can pass something like a DocumentFragment as `rows` to createInstance()? Meaning values are not all toStringed? If so, what types are allowed?

**Comment by justinfagnani** [comment]
@matthewp yes, the key type allowed besides Strings are Nodes, especially TemplateInstance form this proposal.

**Comment by jakearchibald** [comment]
TPAC: consensus for parser changes.

**Comment by wanderview** [comment]
I don't think anyone from mozilla parser team has weighed in yet. cc / @hsivonen

**Comment by jakearchibald** [comment]
Apologies, I just wanted to capture what happened in the room, it wasn't definitive (but I see how it can be read like that).

**Comment by dy** [comment]
That problem can be addressed purely symptomatically with a couple of strong heuristics.
1. If there's insertion `{{ }}` right before the _empty_ table, almost likely it was intended to be inside but parser hoisted it up.
2. If the insertion `{{ }}` contains `<tbody>, <tr>, <th>` etc. - most surely it was supposed to reside in the `<table>`.

These two conditions I believe exclude practical chance user meant to place `<tr>` (or `<thead>` etc.) right before the empty `<table>`.

**Comment by dy** [comment]
As discovered in [templize](https://github.com/spectjs/templize) and [uce-template](https://github.com/WebReflection/uce-template), the simplest workaround is
```js
<template>
  <table>
    <!--{{rows}}-->
  </table>
</template>
```
It has zero cost from performance/code size perspective.
